### ISSUE TITLE
Merging works fails in some cases "expected /type/author, found /type/redirect"

### ISSUE THREAD

**Issue by bitnapper** [internetarchive/openlibrary#8069]
Some Merges do not work, when auther is redirected.

### Evidence / Screenshot (if possible)

`{
  "error": "bad_data",
  "message": "expected /type/author, found /type/redirect",
  "at": {
    "key": "/works/OL2718994W",
    "property": "authors.author"
  },
  "value": "/authors/OL4335630A"
}`

### Relevant url?
https://openlibrary.org/works/merge?records=OL2718994W,OL2719143W,OL10452052W,OL31461757W,OL31461827W,OL31461997W,OL31462067W,OL31465633W&primary=OL2718994W&mrid=64249

### Details

- **Logged in (Y/N)?** y
- **Browser type/version?** Firefox 114.0.2
- **Operating system?** macOS Ventura 13.4.1
- **Environment (prod/dev/local)?** prod

**Comment by LeadSongDog** [comment]
@bitnapper 
Some of those work records show no linked author, which may be a factor

**Comment by bitnapper** [comment]
> @bitnapper 
> 
> Some of those work records (OL31461827W,OL31461997W,OL31462067W) show no linked author, which may be a factor.

You are right. In the merging preview it has an author entry but in the works page it has not. If you set an author before merging it works.

**Comment by bitnapper** [comment]
This one cant be merged because one edition seems to block it:

https://openlibrary.org/works/merge?records=OL2719042W,OL10452151W,OL31461888W,OL31461962W&primary=OL2719042W&mrid=64257

I tried to set the author again on this one but get an error:

https://openlibrary.org/books/OL22832530M/Inspector_West_at_home/edit

`expected /type/author, found /type/redirect`

![image](https://github.com/internetarchive/openlibrary/assets/2772669/ec82741f-f5ec-4086-83d7-269e605765cd)

**Comment by bitnapper** [comment]
- https://openlibrary.org/books/OL19210124M/
- https://openlibrary.org/books/OL22832530M/

**Comment by bitnapper** [comment]
https://openlibrary.org/works/merge?records=OL10581350W,OL11320148W&primary=OL10581350W&mrid=116907

**Comment by LeadSongDog** [comment]
> This one cant be merged because one edition seems to block it:
> 
> https://openlibrary.org/works/merge?records=OL2719042W,OL10452151W,OL31461888W,OL31461962W&primary=OL2719042W&mrid=64257
> 
> I tried to set the author again on this one but get an error:
> 
> https://openlibrary.org/books/OL22832530M/Inspector_West_at_home/edit
> 
> `expected /type/author, found /type/redirect`
> 
> ![image](https://private-user-images.githubusercontent.com/2772669/252279544-ec82741f-f5ec-4086-83d7-269e605765cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc2NTQ4MDAsIm5iZiI6MTcwNzY1NDUwMCwicGF0aCI6Ii8yNzcyNjY5LzI1MjI3OTU0NC1lYzgyNzQxZi1mNWVjLTQwODYtODNkNy0yNjllNjA1NzY1Y2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMTFUMTIyODIwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjM0OGM4NWVkN2NmMzhjZDM0NWYwNGY1NzI3YmY2YzAxNzRkZmE1NmVlMDUyYTNlYWFiODk3NzgwMjhkYzFiNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Ofb2QmN4_tzjVyKmkLviV6L0J8wj1-J-PROI7MzGa4k)

Looking at https://openlibrary.org/works/OL10452151W.json That work has links from two editions that each show a redirected author record, distinct from but resolving to the merged author record OL2751971A found for the work. 

Root cause is that redirects are not being cleaned up from editions after author merges change work records. IMHO, such redirects should simply be pruned from the edition when they resolve to an author listed for the edition’s linked work. I seem to recall an old issue on this matter.

**Comment by LeadSongDog** [comment]
Agree that reporting success and closing should not happen when there is in fact a failure. My point was it should not need to fail: that seems to be the result of having edition records wrongly point to author records, aggravated by those author records being redirects. While I would prefer removing the author linkages from edition records, they should at least resolve to substantive author records, not redirects.
